app/src/query.rs

```diff
@@ -248,10 +249,22 @@ impl<'app, D: Dependencies<'app>> ApplicationReader<'app, D> {
         block_number: BlockNumberOrTag,
     ) -> Result<Vec<u8>> {
         let height = self.resolve_height(block_number)?;
+        let block_header = self
+            .block_queries
+            .by_height(&self.storage, height, false)
+            .map_err(|_| Error::DatabaseState)?
+            .map(|block| HeaderForExecution {
+                number: height,
+                timestamp: block.0.header.timestamp,
+                prev_randao: block.0.header.mix_hash,
+                chain_id: self.genesis_config.chain_id,
+            })
+            .unwrap_or_default();
         call_transaction(
             transaction,
             &self.state_queries.resolver_at(height)?,
             &self.evm_storage,
+            block_header,
```

Comment:
As a side note, why do we add it for `eth_call` only, and not the other view calls? I don't quite see the convenience of this.

Reply:
The header is only required for view calls that trigger execution and there are only two of those: `eth_call` and `eth_estimateGas`. The latter was already correctly using the header in its execution because it follows the same flow as for transactions. `eth_call` needed fixing because it uses a flow separate from transactions which was missing the header.